apps/omg_watcher/lib/utxo_selection.ex

```diff
@@ -0,0 +1,195 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.UtxoSelection do
+  @moduledoc """
+  Provides Utxos selection and merging algorithms.
+  """
+
+  alias OMG.API.Crypto
+  alias OMG.API.State.Transaction
+  alias OMG.Watcher.DB
+
+  require Transaction
+
+  @type payment_t() :: %{
+          owner: Crypto.address_t() | nil,
+          currency: Transaction.currency(),
+          amount: pos_integer()
+        }
+
+  @type fee_t() :: %{
+          currency: Transaction.currency(),
+          amount: non_neg_integer()
+        }
+
+  @type order_t() :: %{
+          owner: Crypto.address_t(),
+          payments: nonempty_list(payment_t()),
+          fee: fee_t(),
+          metadata: binary() | nil
+        }
+
+  @type transaction_t() :: %{
+          inputs: nonempty_list(%DB.TxOutput{}),
+          outputs: nonempty_list(payment_t()),
+          fee: %{amount: integer, currency: Transaction.currency()},
```

Comment:
Should this be `fee_t()` as defined above?

Reply:
Yes, thank you!